cmd/repo-init/main.go

```diff
@@ -462,25 +462,46 @@ func fetchOrDefaultWithPrompt(msg, def string) string {
 	return ""
 }
 
+type RepoProwConfig struct {
+	Tide TideRepoProwConfig `json:"tide,omitempty"`
+}
+
+type TideRepoProwConfig struct {
+	Queries prowconfig.TideQueries `json:"queries,omitempty"`
+}
+
 func updateProwConfig(config initConfig, releaseRepo string) error {
 	configPath := path.Join(releaseRepo, ciopconfig.ConfigInRepoPath)
+	configDir := path.Dir(configPath)
+	tmpDir, err := ioutil.TempDir("", "repo-init")
+	if err != nil {
+		return fmt.Errorf("failed to create temp dir: %w", err)
+	}
+	defer os.Remove(tmpDir)
+
+	targetOrgRepo := prowconfig.OrgRepo{Org: config.Org, Repo: config.Repo}
+	for i, orgRepo := range []string{
+		path.Join(config.Org, config.Repo),
+		path.Join("openshift", "cluster-version-operator"),
+		path.Join("openshift", "ci-tools"),
+	} {
+		src := path.Join(configDir, orgRepo, ciopconfig.SupplementalProwConfigFileName)
+		logrus.WithField("src", src).Info("copying ...")
+		if _, err := os.Stat(src); errors.Is(err, os.ErrNotExist) {
+			continue
+		}
+		if _, err := fileutils.CopyFile(src, path.Join(tmpDir, fmt.Sprintf("%d%s", i, ciopconfig.SupplementalProwConfigFileName))); err != nil {
+			return fmt.Errorf("failed to copy file: %w", err)
+		}
+	}
+
 	agent := prowconfig.Agent{}
-	if err := agent.Start(configPath, "", nil, ""); err != nil {
```

Comment:
The prowconfigs for org/repo have been sharded into its own folders.
But we load only the global one.
So we did not have the config for `openshift/cluster-version-operator` and `openshift/ci-tools`.

This PR copies them to a tmp folder and load them as supplemental configs.

Reply:
Is this done because the cost of loading all the files is prohibitive?

We have `release.go` in `pkg/config/` for shared code that operates on `openshift/release`, would it make sense to put the loading code there so it doesn't diverge again in the future?  It would be preferable to leaving the possibility for a future bug like this, if the cost is not too high.

Reply:
Modification:
- Loading all of them now.
- Moving the loading code to `release.go`.